fastly/director_test.go

```diff
@@ -7,31 +7,71 @@ import (
 func TestClient_Directors(t *testing.T) {
 	t.Parallel()
 
-	var err error
 	var tv *Version
 	record(t, "directors/version", func(c *Client) {
 		tv = testVersion(t, c)
 	})
 
 	// Create
-	var b *Director
+	var (
+		b                  *Backend
+		d                  *Director
+		errBackend         error
+		errDirector        error
+		errDirectorBackend error
+	)
 	record(t, "directors/create", func(c *Client) {
-		b, err = c.CreateDirector(&CreateDirectorInput{
+		b, errBackend = c.CreateBackend(&CreateBackendInput{
+			ServiceID:      testServiceID,
+			ServiceVersion: tv.Number,
+			Name:           "test-backend",
+			Address:        "integ-test.go-fastly.com",
+			Port:           Uint(1234),
+			ConnectTimeout: Uint(1500),
+			OverrideHost:   "origin.example.com",
+			SSLCiphers:     "DHE-RSA-AES256-SHA:DHE-RSA-CAMELLIA256-SHA:AES256-GCM-SHA384",
+		})
+		d, errDirector = c.CreateDirector(&CreateDirectorInput{
 			ServiceID:      testServiceID,
 			ServiceVersion: tv.Number,
 			Name:           "test-director",
 			Quorum:         Uint(50),
 			Type:           DirectorTypeRandom,
 			Retries:        Uint(5),
 		})
+		_, errDirectorBackend = c.CreateDirectorBackend(&CreateDirectorBackendInput{
+			ServiceID:      testServiceID,
+			ServiceVersion: tv.Number,
+			Director:       "test-director",
+			Backend:        b.Name,
```

Comment:
In the test suite I ensure when creating a Director we also create a Backend and then create a relationship between the two.